src/toolchain.rs

```diff
@@ -374,6 +374,37 @@ impl<'a> Toolchain<'a> {
         Ok(None)
     }
 
+    fn maybe_do_rust_analyzer_fallback(&self, binary: &str) -> anyhow::Result<Option<Command>> {
+        if binary != "rust-analyzer" && binary != "rust-analyzer.exe"
+            || self.binary_file("rust-analyzer").exists()
+        {
+            return Ok(None);
+        }
+
+        let proc = self.cfg.process;
+        let Some(path) = proc.var_os("PATH") else {
+            return Ok(None);
+        };
+
+        let me = env::current_exe()?;
+
+        // Try to find the first `rust-analyzer` under the `$PATH` that is both
+        // an existing file and not the same file as `me`, i.e. not a rustup proxy.
+        Ok(env::split_paths(&path).find_map(|mut p| {
+            p.push(binary);
+            let is_external_ra = p.is_file()
+                // We report `true` on `is_same_file()` error to prevent an invalid `p`
+                // from becoming the candidate.
+                && !is_same_file(&me, &p).unwrap_or(true);
+            if !is_external_ra {
+                return None;
+            }
+            let mut ra = Command::new(p);
+            self.set_env(&mut ra);
+            Some(ra)
```

Comment:
Should we add a log here? I think it is quite important to know rustup special-cased this when debugging?

Reply:
I added log lines for both fallbacks, in stderr just in case stdout is somehow relied on.